pulp_container/app/registry.py

```diff
@@ -123,74 +122,144 @@ async def get_tag(self, request):
+                return web.Response(text=raw_data, headers=response_headers)
             else:
                 raise PathNotResolved(tag_name)
+        else:
+            if distribution.remote_id and distribution.pull_through_distribution_id:
+                # check if the content was updated on the remove and stream it back
+                remote = await distribution.remote.acast()
+                relative_url = "/v2/{name}/manifests/{tag}".format(
+                    name=remote.namespaced_upstream_name, tag=tag_name
+                )
+                tag_url = urljoin(remote.url, relative_url)
+                downloader = remote.get_downloader(url=tag_url)
+                try:
+                    response = await downloader.run(
+                        extra_data={"headers": V2_ACCEPT_HEADERS, "http_method": "head"}
+                    )
+                except ClientResponseError:
+                    raise PathNotResolved(path)
+
+                digest = response.headers.get("docker-content-digest")
+                if tag.tagged_manifest.digest != digest:
+                    downloader = remote.get_downloader(url=tag_url)
+                    try:
+                        response = await downloader.run(extra_data={"headers": V2_ACCEPT_HEADERS})
+                    except ClientResponseError as response_error:
+                        if response_error.status == 429:
+                            # the client could request the manifest outside the docker hub pull
+                            # limit; it is necessary to pass this information back to the client
+                            raise HTTPTooManyRequests()
+                        else:
+                            raise PathNotResolved(path)
+
+                    with open(response.path) as content_file:
+                        raw_data = content_file.read()
+
+                    response.artifact_attributes["file"] = response.path
+                    saved_artifact = await save_artifact(response.artifact_attributes)
+
+                    set_guid(generate_guid())
+                    await sync_to_async(dispatch)(
+                        download_image_data,
+                        exclusive_resources=[repository_version.repository],
+                        kwargs={
+                            "repository_pk": repository_version.repository.pk,
+                            "remote_pk": remote.pk,
+                            "manifest_artifact_pk": saved_artifact.pk,
+                            "tag_name": tag_name,
+                        },
+                    )
+
+                    manifest_data = json.loads(raw_data)
+                    media_type = determine_media_type(manifest_data, response)
+                    if media_type not in (MEDIA_TYPE.MANIFEST_LIST, MEDIA_TYPE.INDEX_OCI):
+                        await self.save_manifest_and_blobs(
+                            digest, manifest_data, media_type, remote, repository, saved_artifact
```

Comment:
the digest you got from docker-content-digest header is not reliable because that is not a required header. you should resort to calcualt_digest as in the except branch on line 176